mypyc/genops.py

```diff
@@ -4611,20 +4616,45 @@ def coerce(self, src: Value, target_type: RType, line: int, force: bool = False)
             return tmp
         return src
 
-    def keyword_args_to_positional(self,
-                                   args: Sequence[Value],
-                                   arg_kinds: List[int],
-                                   arg_names: Sequence[Optional[str]],
-                                   sig: FuncSignature) -> List[Optional[Value]]:
-        # NOTE: This doesn't support *args or **kwargs.
+    def native_args_to_positional(self,
+                                  args: Sequence[Value],
+                                  arg_kinds: List[int],
+                                  arg_names: Sequence[Optional[str]],
+                                  sig: FuncSignature,
+                                  line: int) -> List[Optional[Value]]:
+        # TODO: missing_args_to_error_values and coerce_native_call_args should just
+        # be merged into this. Leaving it to a later PR because it will cause a bunch
+        # of genops test churn.
+
         sig_arg_kinds = [arg.kind for arg in sig.args]
         sig_arg_names = [arg.name for arg in sig.args]
         formal_to_actual = map_actuals_to_formals(arg_kinds,
                                                   arg_names,
                                                   sig_arg_kinds,
                                                   sig_arg_names,
                                                   lambda n: AnyType(TypeOfAny.special_form))
-        return [None if len(lst) == 0 else args[lst[0]] for lst in formal_to_actual]
+
+        # Flatten out the arguments, constructing tuples/dicts for
+        # star args when needed.  As mentioned above, we should also
+        # merge the rest of native call argument munging into this.
+        output_args = []  # type: List[Optional[Value]]
+        for idx, lst in enumerate(formal_to_actual):
+            output_arg = None
+            if sig_arg_kinds[idx] == ARG_STAR:
+                pos_args_list = self.primitive_op(new_list_op, [args[i] for i in lst], line)
```

Comment:
Add comment saying something like 'first create a list, then convert it into a tuple'.

Reply:
Actually we should just not create the tuple that way!

Reply:
Actually we should just not create the tuple that way!